src/components/DatePicker/Month/Month.tsx

```diff
@@ -56,6 +57,65 @@ const Month: React.FC<MonthProps> = ({ year, month, onDaySelect, selectedDays, d
   const calculateIsBetween = useCallback(calculatedDayIsBetween, []);
   const calculateSelectedDayFirstOrLast = useCallback(calculateSelectedDayPosition, []);
 
+  const [focusedDay, setFocusedDay] = React.useState(0);
+
+  const calendarRef = React.useRef<HTMLTableElement>(null);
+
+  const numOfDays = currentDay.month(month).year(year).date(1).daysInMonth();
+
+  const { keyboardProps } = useKeyboardEvents({
+    events: {
+      keydown: {
+        onArrowUp: () => {
+          setFocusedDay((prevCursor) => {
+            if (prevCursor - 7 < 0) {
+              return prevCursor;
+            }
+
```

Comment:
Part (or all) of the logic in this component could probably be moved to a separate hook to make this component easier to read, as the useKeyboardEvents is kinda hard to read into